### ISSUE TITLE
RttR crashes when wrongly using LUA

### ISSUE THREAD

**Issue by Spikeone** [Return-To-The-Roots/s25client#1121]
When using the following snippet:
```
function onStart(isFirstStart)     
    AddStaticObject(17,100, 502, 0xFFFF, 1)
end
```
RttR would just crash. The game still checks for correct parameters (e.g. entering ("17","100",502,0xFFFF, 1) results in a wrong parameter error) but when starting the game with correct parameters RttR just crashes. 
There should be a compile error that you can't use that function directly or something like that.

```
function onStart(isFirstStart)     
    rttr:GetWorld():AddStaticObject(17,100, 502, 0xFFFF, 1)
end
```

**Comment by Flamefire** [comment]
Sorry this is not possible. LUA is not compiled. One possible improvement would be to catch all LUA exceptions and stop the game (going back to the map selection or so) on error. But generally this is an implementation error and one cannot be sure what the state would be after some wrong instructions are executed.

>  entering ("17","100",502,0xFFFF, 1) results in a wrong parameter error

When? On execution I guess?

**Comment by Flow86** [comment]
yes we should catch these errors and perhaps even ignore them (i.e print them in the log and show them as a message in the game) but the game should not crash from it

**Comment by Flamefire** [comment]
Ignore them is not a good idea. Any (LUA) exception must terminate the game as it won't be running as it should afterwards.

**Comment by Spikeone** [comment]
@Flamefire 
Argh must have been drunk... Actual line was:
`rttr:GetWorld():AddStaticObject(17,101, 502, file = 0xFFFF, 1)`

And actual error message was: 
`Lua error: ...ke\Saved Games\Return To The Roots\MAPS/Bergschlumpf.lua:42: ')' expected near '='`
upon loading - don't know where I saw the wrong parameter exception, thought it was this one.

Anyway, if it's not possible we can't fix it at all ;) Although a propper crash message would help, e.g. "Unexpected LUA call, terminating RttR" as a normal user might have a hard time finding the error

**Comment by Flow86** [comment]
> Ignore them is not a good idea. Any (LUA) exception must terminate the game as it won't be running as it should afterwards.

yep terminating the game with message. thats what I said)

**Comment by Flamefire** [comment]
Sure. That's basically a try-catch around the run method of the world. We just need to make sure it works with jumping/skipping and terminates the game, not the program. Maybe selectively choose exceptions? E.g. only lua exceptions. So others do terminate the program as we can't be sure if we can recover from those.

**Comment by Spikeone** [comment]
When using SetRestrictedArea with parameter nil or nil, nil the game crashes without a LUA crash but just throws RttrExitException

**Comment by DevOpsOfChaos** [comment]
I checked the original direct-call repro against current `master` (`37e4166c0e2bbe5215a1751666a82cfc709c278f`) with a small temporary Lua regression probe:

```lua
AddStaticObject(17, 100, 502, 0xFFFF, 1)

Current behavior: this throws a LuaExecutionError and does not crash the test process.

Validation:

Built Test_lua locally with Visual Studio 2022 Debug
Ran Test_lua.exe --run_test=LuaTestSuite/World --report_level=short
Result: 1 test case ... passed, 64 assertions ... passed

So the original direct global AddStaticObject(...) case does not seem reproducible on current master anymore. The broader part of this issue may still apply to runtime Lua error handling / terminating the game gracefully instead of the whole program.
